### PR TITLE
chore(tracer)!: remove Tracer.writer

### PR DESCRIPTION
**Pull request by majorgreys**
The `Tracer.writer` is exposed as part of the public API but is in fact an internal structure. This PR moves the writer to a private attribute `_writer`, following the convention in the codebase without exposing this attribute as a property.

### CODE REVIEW

File: tests/contrib/redis/test_redis.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def test_patch_unpatch(self):
     @snapshot()
     def test_opentracing(self):
         """Ensure OpenTracing works with redis."""
-        writer = ddtrace.tracer.writer
+        writer = ddtrace.tracer._writer
```
Comment (majorgreys) [inline tests/contrib/redis/test_redis.py:376]: In all integrations but this case, we can avoid accessing the internal writer directly. To remove this dependency here would require addressing the `FIXME` highlighted below.

**Review by mabdinur — COMMENTED** [review]
Are planning to deprecate tracer.writer on master?

**Comment by majorgreys** [comment]
> Are planning to deprecate tracer.writer on master?

Yes, will put that up next!

**Review by mabdinur — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]

File: releasenotes/notes/chore-writer-private-b1875a1fa3230236.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    ``Tracer.writer`` is removed from the API. To force flushing of buffered traces to the agent, use ``Tracer.flush()`` instead.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/chore-writer-private-b1875a1fa3230236.yaml:4]: Maybe worth mentioning to reach out to us if `writer` is being used for anything else?

**Review by Kyle-Verhoog — COMMENTED** [review]
Just one thought otherwise lgtm!

**Review by mabdinur — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**